### PR TITLE
remove C library in cpp_master branch

### PR DESCRIPTION
**Pull request by ygj6**
- remove C part files
- rename include/msgpack/*.h to include/msgpack/*.hpp
- remove C part in source code

Closes: https://github.com/msgpack/msgpack-c/issues/872

### CODE REVIEW

**Comment by redboltz** [comment]
It seems that some of appveyor builds are failed. Could you fix it?

**Comment by ygj6** [comment]
In the initial [appveyor.yml](https://github.com/msgpack/msgpack-c/blob/6e7deb809120881634b3ca895e66b2a946084f34/appveyor.yml#L14-L20), when `msvc` is "Visual Studio 10 2010" or "Visual Studio 11 2012" or "Visual Studio 12 2013", the value of `MSGPACK_BOOST` is OFF. If it is set to ON, the compilation will fail because `Boost` cannot be found.
This shows that for the three versions of `msvc`, `Boost` is not supported.
So, should I remove these [three matrices](https://github.com/ygj6/msgpack-c/blob/2f5b13799743f800655680b8eab327c739aad2b5/appveyor.yml#L13-L18) in this PR?

**Comment by redboltz** [comment]
Boost said that MSVC 8.0 SP1, MSVC 9.0 and newer are supported.
https://www.boost.org/doc/libs/1_73_0/libs/log/doc/html/log/installation.html

And https://en.wikipedia.org/wiki/Microsoft_Visual_C%2B%2B#Internal_version_numbering said that  Visual Studio 10 2010 has MSVC++ 10.

So boost supports Visual Studio 10 2010.

I'm not sure appveyor's image management. It seems that appveyor or cmake's problem.
We need to try to find boost on these environment.
I think that we need only header only libraries except some of examples. Install (just place boost image) manually could be an option.

File: CMakeLists.txt [hunk 1]
```diff
@@ -88,37 +72,32 @@ ENDIF ()
...
 OPTION (MSGPACK_BUILD_EXAMPLES "Build msgpack examples." ON)
 
-IF (MSGPACK_BOOST)
-   SET (CMAKE_CXX_FLAGS "-DMSGPACK_USE_BOOST ${CMAKE_CXX_FLAGS}")
-   SET (Boost_USE_MULTITHREADED      ON)
-   SET (Boost_USE_STATIC_RUNTIME    OFF)
-   FIND_PACKAGE (Boost REQUIRED COMPONENTS chrono context system timer)
+SET (Boost_USE_MULTITHREADED      ON)
+SET (Boost_USE_STATIC_RUNTIME    OFF)
+FIND_PACKAGE (Boost REQUIRED COMPONENTS chrono context system timer)
```
Comment (redboltz) [inline CMakeLists.txt:86]:
```suggestion
FIND_PACKAGE (Boost REQUIRED)
```

When I removed components, then cmake find boost correctly on appveyor old msvc.
I think that omitting components is accepted solution.

**Comment by redboltz** [comment]
Some of examples require (not header only) boost library. But old msvc on appveyor doesn't have built library. So older msvc, disabling examples is acceptable solution.
It is better than removing build from the matrix because tests are built and executed.

Could you update appveyor.yml as follows ?


appveyor.yml

```yaml
  matrix:
    - cpp11: -DMSGPACK_CXX11=OFF
      example:  -DMSGPACK_BUILD_EXAMPLES=OFF
      msvc: '"Visual Studio 10 2010"'
    - cpp11: -DMSGPACK_CXX11=OFF
      example:  -DMSGPACK_BUILD_EXAMPLES=OFF
      msvc: '"Visual Studio 11 2012"'
    - cpp11: -DMSGPACK_CXX11=OFF
      example:  -DMSGPACK_BUILD_EXAMPLES=OFF
      msvc: '"Visual Studio 12 2013"'
    - cpp11: -DMSGPACK_CXX11=ON
      example:  -DMSGPACK_BUILD_EXAMPLES=ON
      msvc: '"Visual Studio 14 2015"'
    - cpp11: -DMSGPACK_CXX11=OFF
      example:  -DMSGPACK_BUILD_EXAMPLES=ON
      msvc: '"Visual Studio 14 2015"'
```

```yaml
- cmake -G %msvc% %cpp11% %example% %x3_parse% -DGTEST_LIBRARY=%APPVEYOR_BUILD_FOLDER%\googletest-release-1.7.0\build\Release\gtest.lib -DGTEST_MAIN_LIBRARY=%APPVEYOR_BUILD_FOLDER%\googletest-release-1.7.0\build\Release\gtest_main.lib -DGTEST_INCLUDE_DIR=%APPVEYOR_BUILD_FOLDER%\googletest-release-1.7.0\include -DZLIB_LIBRARY=%APPVEYOR_BUILD_FOLDER%\zlib-1.2.11\build\Release\zlib.lib -DZLIB_INCLUDE_DIR=%APPVEYOR_BUILD_FOLDER%\zlib-1.2.11 -DCMAKE_CXX_FLAGS='"/D_VARIADIC_MAX=10 /EHsc"' ..
```

File: CMakeLists.txt (hunk 1 above)
Comment (ygj6) [reply CMakeLists.txt]: Looks like `COMPONENTS` is necessary on Ubuntu and Mac.

**Comment by redboltz** [comment]
It seems that the fix solves first 3 appveyor builds but unexpected side effects.
Required more fix...

**Comment by redboltz** [comment]
It seems that 5906655 have some error on appveyor but I think that your approach (replacing CMakeLists.txt) is good :)
Text processing on Windows is very difficult to me :(

I will wait your update.

**Comment by redboltz** [comment]
Thank you for updating !

**Pull request merged**